var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -128,9 +136,18 @@ def patch(self):
                     # and using LD_LIBRARY_PATH throughout Spack.
                     f.write('[mkl]\n')
                     f.write('mkl_libs=%s\n'     % 'mkl_rt')
+                    writeLibraryDirs(f, dirs)
                 elif '^atlas' in spec:
                     f.write('[atlas]\n')
                     f.write('atlas_libs=%s\n'   % names)
+                    writeLibraryDirs(f, dirs)
+                elif '^netlib-lapack' in spec:
+                    # netlib requires blas and lapack listed
+                    # separately so that scipy can find them
+                    for library in ['blas', 'lapack']:
```

Comment:
I would add:
```python
if '+' + library in spec:
```
after this line. At this point in the code, we know that either `+blas` or `+lapack` is true but we don't know if both are true. Honestly, idk why we even have variants for this, do people actually build `py-numpy` without BLAS/LAPACK support??

Reply:
@jrood-nrel I think this comment still needs to be addressed.

Another thought: does this support `netlib-lapack+external-blas`? It's possible that the `netlib-lapack` in the spec only provides LAPACK, not BLAS. Theoretically, someone could mix different BLAS and LAPACK providers, correct? Do we need to handle this situation? There are many more BLAS providers than there are LAPACK providers, although I'm not sure how many of these numpy/scipy support:
```console
$ spack providers blas lapack
blas:
atlas  blis  blis  essl  intel-mkl  intel-parallel-studio  netlib-lapack  netlib-xblas  openblas  veclibfort

lapack:
atlas  intel-mkl  intel-parallel-studio  netlib-lapack  openblas  veclibfort
```

Reply:
I see. I never even realized there were variants in `py-numpy` for `blas` and `lapack`, haha.

In regards to `netlib-lapack+external-blas`, it seems like this should work because
```
        if '+blas' in spec:
            lapackblas += spec['blas'].libs
```
gets libraries from the blas provider and not `netlib-lapack` directly. With my new commit it should omit `blas` and/or `lapack` depending on the variants. However, I think the variants should almost just be removed.

Reply:
I think the new commit should resolve this.